threadfix_converter.py

```diff
@@ -1,135 +1,111 @@
-#!/usr/local/bin/python3
+#!/usr/bin/env python3
 
-import datetime
+from datetime import datetime
 import json
 import sys
 
 
-# Converts SARIF to Threadfix
-def make_ref_rule_dict(item):
-    ret_val = {}
-    current_rule = {}
-    cwe_start = 'external/cwe/cwe-'
-
-    current_rule['id'] = item['rule']['id']
-    current_rule['shortDescription'] = item['rule']['description']
-    current_rule['fullDescription'] = item['rule']['description']  # To revisit
-
-    # Severities
-    if 'security_severity_level' in item['rule']:
-        current_rule['nativeSeverity'] = item['rule']['security_severity_level'].capitalize()
-        current_rule['severity'] = item['rule']['security_severity_level'].capitalize()
-    else:
-        current_rule['nativeSeverity'] = item['rule']['severity'].capitalize()
-        current_rule['severity'] = item['rule']['severity'].capitalize()
-
-    vuln_type = item['rule']['id']
-    current_rule['isCwe'] = False
-    for tag in item['rule']['tags']:
-        if str(tag).startswith(cwe_start):
-            vuln_type = str(tag)[len(cwe_start):len(str(tag))]
-            current_rule['isCwe'] = True
-    current_rule['type'] = vuln_type
-
-    ret_val[item['rule']['id']] = current_rule
-
-    return (ret_val)
-
-
-def assemble_findings_for_run(finding_list, item, repo):
+def make_ref_rule_dict(item: dict) -> dict:
+    """
+    Convert SARIF to Threadfix
```

Comment:
I think we should rephrase that. We're technically pulling an array with data and we're not working directly with a SARIF file, so maybe we could say: "convert data to Threadfix" or similar

Reply:
@nbirnel the rest look great! Thank for refactoring the code!